test_weather_data_fetcher.py

```diff
@@ -0,0 +1,61 @@
+import os
+import dotenv
+from unittest import mock
+
+from weather_data_fetcher import fetch_location_coords, fetch_weather_data, get_api_key
+
+
+class TestWeatherDataFetcher:
+    def test_fetch_coords(self):
+        coords_berlin = fetch_location_coords("Berlin")
```

Comment:
You forgot to mock the real API call underneath. Please do that otherwise I cannot run this test without internet.